File: SWP.Core/Entities/ServiceTicket.cs
using SWP.Core.GMSAttribute;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SWP.Core.Entities
{
    [GMSTableName("service_ticket")]
    public class ServiceTicket
    {
        [Column("service_ticket_id")]
        [Key]
        public Guid ServiceTicketId { get; set; }

        [Column("booking_id")]
        public Guid? BookingId { get; set; }

        [Column("vehicle_id")]
        [Required]
        public Guid VehicleId { get; set; }

        [Column("created_by")]
        [Required]
        public Guid CreatedBy { get; set; }

        [Column("created_date")]
        public DateTime? CreatedDate { get; set; }

        [Column("modified_by")]
        public Guid? ModifiedBy { get; set; }

        [Column("modified_date")]
        public DateTime? ModifiedDate { get; set; }

        [Column("service_ticket_status")]
        public byte? ServiceTicketStatus { get; set; }

        [Column("initial_issue")]
        public string? InitialIssue { get; set; }

        [Column("service_ticket_code")]
        public string? ServiceTicketCode { get; set; }

        // Navigation properties
        public Booking? Booking { get; set; }
        public Vehicle Vehicle { get; set; } = null!;
        public User CreatedByUser { get; set; } = null!;
        public User? ModifiedByUser { get; set; }
        public ICollection<ServiceTicketDetail> ServiceTicketDetails { get; set; } = new List<ServiceTicketDetail>();
        public ICollection<TechnicalTask> TechnicalTasks { get; set; } = new List<TechnicalTask>();
        public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
    }
}

File: GarageProject-BE-Springboot-MainFlow/src/main/java/fptu/edu/vn/training/entity/ServiceTicket.java
package fptu.edu.vn.training.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.LinkedHashSet;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "service_ticket")
public class ServiceTicket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "service_ticket_id")
    private Integer serviceTicketId;

    @Column(name = "booking_id")
    private Integer bookingId;

    @Column(name = "vehicle_id", nullable = false)
    private Integer vehicleId;

    @Column(name = "created_by", nullable = false)
    private Integer createdBy;

    @Column(name = "created_date")
    private LocalDateTime createdDate;

    @Column(name = "modified_by")
    private Integer modifiedBy;

    @Column(name = "modified_date")
    private LocalDateTime modifiedDate;

    @Column(name = "service_ticket_status")
    private Byte serviceTicketStatus;

    @Column(name = "initial_issue")
    private String initialIssue;

    @Column(name = "service_ticket_code", length = 20)
    private String serviceTicketCode;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Integer isDeleted = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vehicle_id", insertable = false, updatable = false)
    private Vehicle vehicle;

    @Builder.Default
    @OneToMany(mappedBy = "serviceTicket", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = false)
    private Set<ServiceTicketDetail> serviceTicketDetails = new LinkedHashSet<>();
}
